### PR TITLE
feat(trusted): add a permanent agent VM

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3486

creating the permanent Agent VM used for trusted.ci

### CODE REVIEW

File: trusted.ci.jenkins.io.tf [hunk 1]
```diff
@@ -21,7 +26,11 @@ resource "azurerm_resource_group" "trusted_ci_jenkins_io_controller" {
   location = var.location
   tags     = local.default_tags
 }
-
+resource "azurerm_resource_group" "trusted_ci_jenkins_io_permanent_agents" {
```
Comment (lemeurherve) [inline trusted.ci.jenkins.io.tf:29]: Is a new resource group needed?

Why not putting it in the existing "trusted" resource groupe?

File: trusted.ci.jenkins.io.tf [hunk 2]
```diff
@@ -209,6 +218,72 @@ resource "azurerm_private_dns_a_record" "trusted_ci_controller" {
...
+  }
+}
+
+resource "azurerm_managed_disk" "trusted_permanent_agent_data_disk" {
+  name                 = "trusted-permanent-agent-data-disk"
+  location             = azurerm_resource_group.trusted_ci_jenkins_io_permanent_agents.location
+  resource_group_name  = azurerm_resource_group.trusted_ci_jenkins_io_permanent_agents.name
+  storage_account_type = "Standard_LRS"
+  create_option        = "Empty"
+  disk_size_gb         = "1024" # no intermediate between 512 and 1024
```
Comment (lemeurherve) [inline trusted.ci.jenkins.io.tf:274]: 512 is not enough?

File: trusted.ci.jenkins.io.tf (hunk 1 above)
Comment (smerle33) [reply trusted.ci.jenkins.io.tf:29]: as we had a group for the ephemeral, we decided at the beginning to have 3, I would rather have only one, the trusted one ... will talk this with @dduportal  next week and to a subsequent PR if needed.

File: trusted.ci.jenkins.io.tf (hunk 2 above)
Comment (smerle33) [reply trusted.ci.jenkins.io.tf:274]: a little short : /dev/xvdf1      591G  388G  176G  69% /home/jenkins
we might cross the 80% level (512Go will be seen as 500) ... I am not sure we want to take the risk during a `trusted` process, if it had a lot of data during the pipeline ....

**Review by timja — APPROVED** [review]

File: trusted.ci.jenkins.io.tf [hunk 3]
```diff
@@ -209,6 +218,72 @@ resource "azurerm_private_dns_a_record" "trusted_ci_controller" {
...
+
+  tags = local.default_tags
+}
+
+resource "azurerm_virtual_machine_data_disk_attachment" "trusted_permanent_agent_data_disk" {
```
Comment (timja) [inline trusted.ci.jenkins.io.tf:279]: missing dns record for the agent?

Comment (smerle33) [reply trusted.ci.jenkins.io.tf:279]: YES thanks @timja

Comment (smerle33) [reply trusted.ci.jenkins.io.tf:279]: should be ok now for the A record, if I understood correctly the documentation.

**Review by timja — APPROVED** [review]

File: trusted.ci.jenkins.io.tf (hunk 1 above)
Comment (dduportal) [reply trusted.ci.jenkins.io.tf:29]: As per https://github.com/jenkins-infra/helpdesk/issues/3486#issuecomment-1540067039, issue updated to explain why.

TL;DR; different lifecycles driven by adding locks (o not).
